### ISSUE TITLE
[FR] Color UI: improve numeric editors, adding more buttons, enable cancel

### ISSUE THREAD

**Issue by vlsi** [MarlinFirmware/Marlin#26230]
### Is your feature request related to a problem? Please describe.

It looks like the current editors for "probe offset", "drive current", "preheat temperature" are hard to use.
In my MKS_ROBIN_TFT35 (480x320, 3.5") the editors are just a "volume control" with two buttons "plus, minus, ok".

Suppose the drive current is "800ma". How could I alter it to 400? "minus" button would require many clicks (e.g. 40 if each reduces 10ma), and the volume controller does not help much as it is inaccurate.

### Describe the feature you want

I suggest the following UI:

```
E0 current, ma, 
   [ ^ ] [ ^ ] [ ^ ]
     8     0     0
   [ v ] [ v ] [ v ]

[ ] apply immediately
[apply] [cancel]
```

```
Probe Z offset, mm
   [ ^ ]   [ ^ ] [ ^ ]
 -   0   .   9     3
   [ v ]   [ v ] [ v ]

[ ] apply immediately
[apply] [cancel]
```

```
Fan speed, %
   [ ^ ] [ ^ ] [ ^ ]
           7     0
   [ v ] [ v ] [ v ]

[ ] apply immediately
[apply] [cancel]
```

In other words:
1) For each decimal place, there could be a button to increase/decrease. The button would be big enough so one could perform fine-tuning while watching the print
2) There could be "apply immediately" checkbox, so values like "fan speed" could be adjusted on the fly without forcing the user to click "ok" to apply the setting
3) There should be "cancel" button so the edit could be discarded if it is not wanted

**Comment by thisiskeithb** [comment]
This is being worked on:

- https://github.com/MarlinFirmware/Marlin/pull/25649
